delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/IriOrBNode.scala

```diff
@@ -176,6 +176,18 @@ object IriOrBNode {
     override val asIri: Option[Iri] = Some(this)
 
     override val asBNode: Option[BNode] = None
+
+    /**
+      * Returns a new absolute Iri resolving the current relative [[Iri]] with the passed absolute [[Iri]]
+      */
+    def resolvedAgainst(iri: Iri): Iri =
```

Comment:
this seems to fail silently if the provided iri is not absolute; I'm wondering if it would make sense to have a `RelativeIri` wrapper with a safe constructor to avoid these situations.

Reply:
It is the same logic than the akka http `Uri`. It does not fail silently, there is nothing to resolve if the iri is already resolved. That's the desired behaviour.

Reply:
and are we happy with the possibility of ending up with a relative iri?

Reply:
The tests will tell you :) In our code that won't happen cause we are always using the `BaseUri` for this. I don't feel like introducing a `AbsoluteIri` thing because that change is going to cascade all over the place now